### DISCUSSION TITLE
How to Read System.Byte[ ] as a string

### DISCUSSION THREAD

**Discussion by dvgmdvgm** [vfsfitvnm/frida-il2cpp-bridge#117] [Q&A]
Hello!
My dumped constructor looks like `System.Void .ctor(System.Byte[] hash);`
How i can read this "hash" as String..
Just can read like `hash= [538,193,96,452,264,193,556,683,1,39]`
But it's should be like that `hash= A*bF/&ssKq`

Can you help me please?

**Comment by vfsfitvnm** [comment]
@CODEDAGOTH 
Well, `[538,193,96,452,264,193,556,683,1,39]` does not contain bytes, whose range is 0-255 (did you hardcode that?)...

There are several ways to accomplish your task:
 ```ts
 Il2Cpp.perform(() => {
    const SystemByte = Il2Cpp.Image.corlib.classes["System.Byte"];

    const arr = Il2Cpp.Array.from(SystemByte, [0x68, 0x65, 0x6c, 0x6c, 0x6f]);
    console.log(arr);
    
    // Use any of the existing C# methods
    const SystemParameterizedStrings = Il2Cpp.Image.corlib.classes["System.ParameterizedStrings"];
    const str0 = SystemParameterizedStrings.methods.StringFromAsciiBytes.invoke<Il2Cpp.String>(arr, 0, arr.length);
    console.log(str0);

    // Use JS
    const str1 = String.fromCharCode(...arr);
    console.log(str1);
});
```

**Comment by dvgmdvgm** [comment]
@vfsfitvnm yeah.. i'm just type random numbers []..
Anyway your code works great and i could to use only this one line
`const str1 = String.fromCharCode(...arr);`
This is enouth) Thank you
